Civi/Token/TokenRow.php

```diff
@@ -144,31 +144,35 @@ public function context($a = NULL, $b = NULL) {
   /**
    * Update the value of a token.
    *
-   * If you are reading this it probably means you can't follow this function.
+   * Eileen said: If you are reading this it probably means you can't follow this function.
    * Don't worry - I've stared at it & all I see is a bunch of letters. However,
    * the answer to your problem is almost certainly that you are passing in null
    * rather than an empty string for 'c'.
+   * MJW said: I've renamed the bunch of letters so they might be a bit more meaningful.
+   * Also, I don't think this function should ever be called without $tokenField being set
+   * but tests do call it like that.
+   *
+   * @param string|array $tokenEntity
+   * @param string|array $tokenField
+   * @param string|array $tokenValue
```

Comment:
Actually `$tokenValue` can also be `\DateTime` and `\Brick\Money\Money`. Those objects are handled in [TokenProcessor](https://github.com/civicrm/civicrm-core/blob/161e790ebd2c41ca651db8b2df8c335d4fe2264f/Civi/Token/TokenProcessor.php#L480).

BTW: I'm wondering what kind of arrays are allowed and what the expected result is.

Reply:
PR to fix this: #31584.